core/trino-main/src/main/java/io/trino/FeaturesConfig.java

```diff
@@ -479,4 +482,9 @@ public FeaturesConfig setForceSpillingJoin(boolean forceSpillingJoin)
         this.forceSpillingJoin = forceSpillingJoin;
         return this;
     }
+
+    public void applyFaultTolerantExecutionDefaults()
+    {
+        exchangeCompressionEnabled = true;
+    }
```

Comment:
while it's good to keep FTE defaults in same place as the other configs the method being public can lead to misuse. Config objects from Airlift are immutable so using the method anywhere except `bindConfigDefaults` would not have intended effects.

Would declaring a constant here and using it in `bindConfigDefaults` be better in that regards?

Reply:
Does Airlift do anything special with `set*` methods?

Reply:
You can obviously call `set*` after the fact but since the config is Guice managed and also affects other modules via conditional modules the point in time at which the `set*` is called can have different impact and makes reasoning about whole thing problematic.

An example is https://github.com/trinodb/trino/pull/14921 (not related to modifying the config object but regarding the effect when the config is modified can have).